Backport jetpack 15.9-beta changes (#49464)
Automattic/jetpack: projects/packages/newsletter/CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.9.0] - 2026-06-08
### Added
- Add Jetpack branding to the Daily Writing Prompt dashboard widget. [#49438]
- Add the Daily Writing Prompt dashboard widget, moved here from the jetpack-mu-wpcom package. [#49425]

### Changed

> - Remove the per-page Hello Dolly rule; its content is now covered by the centralized normalize rule shipped with `@automattic/jetpack-components`'s AdminPage component. [#48472]

Reviewer: ```suggestion
- Remove the per-page Hello Dolly rule, as it is now handled by `@automattic/jetpack-components`'s AdminPage component. [#48472]
```

### Fixed
- Include the routes directory in the TypeScript config so dashboard route files type-check correctly. [#49368]

> - Don't show the WordPress.com connection prompt on Simple sites, which are already connected. [#49365]

Reviewer: I personally try to avoid contractions, particularly as the initial verb. Easy enough to switch "don't show" to "hide":
```suggestion
- Hide the WordPress.com connection prompt on Simple sites, which are already connected. [#49365]
```

- Restore body padding by wrapping content in Dialog.Content in Add subscribers and Comp modals. [#49327]

> - Subscribers Dashboard: Fix several Comp modal issues — show the plan billing interval in the picker, correct plan selection, disable and explain (via tooltip) plans the reader already has, surface the specific already-comped error message, and fix a clipped focus ring. [#49441]

Reviewer: I'd just go with this, but feel free to ignore this one.
```suggestion
- Subscribers Dashboard: Fix several Comp modal issues. [#49441]
```
